splunk-otel-android/src/main/java/com/splunk/rum/volley/VolleyHttpClientAttributesExtractor.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.rum.volley;
+
+import androidx.annotation.Nullable;
+
+import com.android.volley.AuthFailureError;
+import com.android.volley.Header;
+import com.android.volley.Request;
+import com.android.volley.toolbox.HttpResponse;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import io.opentelemetry.instrumentation.api.instrumenter.http.CapturedHttpHeaders;
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpClientAttributesExtractor;
+
+final class VolleyHttpClientAttributesExtractor extends HttpClientAttributesExtractor<RequestWrapper, HttpResponse> {
+
+    VolleyHttpClientAttributesExtractor(CapturedHttpHeaders capturedHttpHeaders) {
+        super(capturedHttpHeaders);
+    }
+
+    @Override
+    protected String url(RequestWrapper requestWrapper) {
+        return requestWrapper.getRequest().getUrl();
+    }
+
+    @Nullable
+    @Override
+    protected String flavor(RequestWrapper requestWrapper, @Nullable HttpResponse response) {
+        return null;
+    }
+
+    @Override
+    protected String method(RequestWrapper requestWrapper) {
+        Request<?> request = requestWrapper.getRequest();
+        switch (request.getMethod()) {
+            case -1:
+                try {
```

Comment:
I would probably just return `null` in this case. It feels like calling `getPostBody` could have side effects, perhaps it is too risky to call?